crates/pop-common/src/git.rs

```diff
@@ -29,8 +29,12 @@ impl Git {
 		};
 
 		if let Some(reference) = reference {
-			let object = repo.revparse_single(reference).expect("Object not found");
-			repo.checkout_tree(&object, None).expect("Failed to checkout");
+			let object = repo
```

Comment:
This fix was necessary to enable cloning from a specific branch (which previously didn’t work). It can be submitted as a separate PR to main.